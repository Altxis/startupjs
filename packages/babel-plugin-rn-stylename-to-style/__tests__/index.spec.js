const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index.cjs')
const packageJson = require('../package.json')

const { name: pluginName } = packageJson

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    extensions: ['styl', 'css']
  },
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  tests: {
    'Regular string': /* js */`
      import './index.styl'
      function Test () {
        return (
          <div styleName='root active'>
            <span styleName='title'>Title</span>
            <span styleName='description'>Description</span>
            <button styleName='submit disabled'>Submit</button>
          </div>
        )
      }
    `,
    'Regular string with existing style': /* js */`
      import './index.styl'
      function Test ({ style }) {
        const titleStyle = { color: 'red', fontWeight: 'bold' }
        return (
          <div style={[style, { backgroundColor: 'black' }]} styleName='root active'>
            <span styleName='title' style={titleStyle}>Title</span>
            <span styleName='description' style={{ color: 'green' }}>Description</span>
            <button styleName='submit disabled'>Submit</button>
          </div>
        )
      }
    `,
    'Arrays and objects': /* js */`
      import './index.styl'
      function Test ({ style, active, submit, disabled }) {
        const titleStyle = { color: 'red', fontWeight: 'bold' }
        return (
          <div style={style} styleName={['root', {active}]}>
            <span styleName={['title']} style={titleStyle}>Title</span>
            <span styleName={['description']} style={{ color: 'green' }}>Description</span>
            <button style={{ color: 'pink' }} styleName={{submit, disabled}}>Submit</button>
          </div>
        )
      }
    `,
    'Puts compiled attribute to the end of attributes list': /* js */`
      import './index.styl'
      function Test ({ style, active, submit, disabled }) {
        return (
          <div style={style} styleName={['root', {active}]} title='container'>
            <span styleName='text' style={{ color: 'red' }} center title='text' />
          </div>
        )
      }
    `,
    'With ::part() styles': /* js */`
      import './index.styl'
      function Test ({ style, active, submit, disabled, titleStyle }) {
        return (
          <Card style={style} styleName={['root', {active}]} titleStyle={[titleStyle, { color: 'green' }]} title='container'>
            <Content styleName='text' style={{ color: 'red' }} footerStyle={{ color: 'blue' }} center title='text' />
          </Card>
        )
      }
    `,
    'DEPRECATED! With ::part() styles and deprecated *StyleName props': /* js */`
      import './index.styl'
      function Test ({ style, active, submit, disabled, titleStyle }) {
        return (
          <Card style={style} styleName={['root', {active}]} titleStyleName={['title', {active}]} titleStyle={[titleStyle, { color: 'green' }]} title='container'>
            <Content styleName='text' style={{ color: 'red' }} footerStyleName='footer' footerStyle={{ color: 'blue' }} center title='text' />
          </Card>
        )
      }
    `,
    'part attribute. No props': /* js */`
      import './index.styl'
      function Test () {
        return (
          <Card
            part='card'
            style={{ color: 'blue' }}
            titleStyle={{ color: 'red' }}
          >
            <Content
              styleName='content'
              part='content active'
              style={{ fontWeight: 'bold' }}
              bodyStyle={{ color: 'blue' }}
            />
          </Card>
        )
      }
    `,
    'part attribute. With destructed props': /* js */`
      import './index.styl'
      function Test ({ style, cardStyle, title }) {
        return (
          <Card
            part='card'
            style={{ color: 'blue' }}
            titleStyle={{ color: 'red' }}
          >
            <Content part='content active' />
          </Card>
        )
      }
    `,
    'part attribute. With named props': /* js */`
      import './index.styl'
      function Test (props) {
        return (
          <Card
            part='card'
            style={{ color: 'blue' }}
            titleStyle={{ color: 'red' }}
          >
            <Content part='content active' />
          </Card>
        )
      }
    `,
    'part attribute. With anon function within named fn': /* js */`
      import './index.styl'
      const Test = ({ style, cardStyle: myCardStyle, contentStyle, title }) => {
        function render () {
          return (
            <Card
              part='card'
              style={{ color: 'blue' }}
              titleStyle={{ color: 'red' }}
            >
              <Content part='content active' />
            </Card>
          )
        }
        return render()
      }
    `,
    'magic \'root\' part': /* js */`
      import './index.styl'
      function Test ({ title }) {
        return (
          <Card
            part='root'
            style={{ color: 'blue' }}
            titleStyle={{ color: 'red' }}
          >
            <Content part='content active' />
          </Card>
        )
      }
    `,
    'magic \'root\' part with existing style prop': /* js */`
      import './index.styl'
      function Test ({ title, style }) {
        return (
          <Card
            part='root'
            style={{ color: 'blue' }}
            titleStyle={{ color: 'red' }}
          >
            <Content part='content active' />
          </Card>
        )
      }
    `
  }
})
