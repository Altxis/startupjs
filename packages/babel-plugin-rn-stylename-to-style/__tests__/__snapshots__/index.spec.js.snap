// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-rn-stylename-to-style Arrays and objects: Arrays and objects 1`] = `

import './index.styl'
function Test ({ style, active, submit, disabled }) {
  const titleStyle = { color: 'red', fontWeight: 'bold' }
  return (
    <div style={style} styleName={['root', {active}]}>
      <span styleName={['title']} style={titleStyle}>Title</span>
      <span styleName={['description']} style={{ color: 'green' }}>Description</span>
      <button style={{ color: 'pink' }} styleName={{submit, disabled}}>Submit</button>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _temp from './index.styl'

var _processStyleName = require('@startupjs/babel-plugin-rn-stylename-to-style/process').process

function Test({ style, active, submit, disabled }) {
  const titleStyle = {
    color: 'red',
    fontWeight: 'bold'
  }
  return (
    <div
      {..._processStyleName(
        [
          'root',
          {
            active
          }
        ],
        _temp,
        {
          style: style
        }
      )}
    >
      <span
        {..._processStyleName(['title'], _temp, {
          style: titleStyle
        })}
      >
        Title
      </span>
      <span
        {..._processStyleName(['description'], _temp, {
          style: {
            color: 'green'
          }
        })}
      >
        Description
      </span>
      <button
        {..._processStyleName(
          {
            submit,
            disabled
          },
          _temp,
          {
            style: {
              color: 'pink'
            }
          }
        )}
      >
        Submit
      </button>
    </div>
  )
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-to-style Puts compiled attribute to the end of attributes list: Puts compiled attribute to the end of attributes list 1`] = `

import './index.styl'
function Test ({ style, active, submit, disabled }) {
  return (
    <div style={style} styleName={['root', {active}]} title='container'>
      <span styleName='text' style={{ color: 'red' }} center title='text' />
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _temp from './index.styl'

var _processStyleName = require('@startupjs/babel-plugin-rn-stylename-to-style/process').process

function Test({ style, active, submit, disabled }) {
  return (
    <div
      title="container"
      {..._processStyleName(
        [
          'root',
          {
            active
          }
        ],
        _temp,
        {
          style: style
        }
      )}
    >
      <span
        center
        title="text"
        {..._processStyleName('text', _temp, {
          style: {
            color: 'red'
          }
        })}
      />
    </div>
  )
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-to-style Regular string with existing style: Regular string with existing style 1`] = `

import './index.styl'
function Test ({ style }) {
  const titleStyle = { color: 'red', fontWeight: 'bold' }
  return (
    <div style={[style, { backgroundColor: 'black' }]} styleName='root active'>
      <span styleName='title' style={titleStyle}>Title</span>
      <span styleName='description' style={{ color: 'green' }}>Description</span>
      <button styleName='submit disabled'>Submit</button>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _temp from './index.styl'

var _processStyleName = require('@startupjs/babel-plugin-rn-stylename-to-style/process').process

function Test({ style }) {
  const titleStyle = {
    color: 'red',
    fontWeight: 'bold'
  }
  return (
    <div
      {..._processStyleName('root active', _temp, {
        style: [
          style,
          {
            backgroundColor: 'black'
          }
        ]
      })}
    >
      <span
        {..._processStyleName('title', _temp, {
          style: titleStyle
        })}
      >
        Title
      </span>
      <span
        {..._processStyleName('description', _temp, {
          style: {
            color: 'green'
          }
        })}
      >
        Description
      </span>
      <button {..._processStyleName('submit disabled', _temp, {})}>Submit</button>
    </div>
  )
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-to-style Regular string: Regular string 1`] = `

import './index.styl'
function Test () {
  return (
    <div styleName='root active'>
      <span styleName='title'>Title</span>
      <span styleName='description'>Description</span>
      <button styleName='submit disabled'>Submit</button>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _temp from './index.styl'

var _processStyleName = require('@startupjs/babel-plugin-rn-stylename-to-style/process').process

function Test() {
  return (
    <div {..._processStyleName('root active', _temp, {})}>
      <span {..._processStyleName('title', _temp, {})}>Title</span>
      <span {..._processStyleName('description', _temp, {})}>Description</span>
      <button {..._processStyleName('submit disabled', _temp, {})}>Submit</button>
    </div>
  )
}


`;

exports[`@startupjs/babel-plugin-rn-stylename-to-style With ::part() styles: With ::part() styles 1`] = `

import './index.styl'
function Test ({ style, active, submit, disabled, titleStyle }) {
  return (
    <Card style={style} styleName={['root', {active}]} titleStyle={[titleStyle, { color: 'green' }]} title='container'>
      <Content styleName='text' style={{ color: 'red' }} footerStyle={{ color: 'blue' }} center title='text' />
    </Card>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _temp from './index.styl'

var _processStyleName = require('@startupjs/babel-plugin-rn-stylename-to-style/process').process

function Test({ style, active, submit, disabled, titleStyle }) {
  return (
    <Card
      title="container"
      {..._processStyleName(
        [
          'root',
          {
            active
          }
        ],
        _temp,
        {
          style: style,
          titleStyle: [
            titleStyle,
            {
              color: 'green'
            }
          ]
        }
      )}
    >
      <Content
        center
        title="text"
        {..._processStyleName('text', _temp, {
          style: {
            color: 'red'
          },
          footerStyle: {
            color: 'blue'
          }
        })}
      />
    </Card>
  )
}


`;
